### PR TITLE
Track modes rebase

### PR DESCRIPTION
**Pull request by gkatsev**
Rebase #57.

### CODE REVIEW

File: test/ads.html [hunk 1]
```diff
@@ -5,13 +5,14 @@
   <title>ads Test Suite</title>
   <!-- Load local video.js -->
   <link rel="stylesheet" href="../node_modules/video.js/dist/video-js/video-js.min.css" media="screen">
-  <script src="../node_modules/video.js/dist/video-js/video.js"></script>
+  <script src="../node_modules/video.js/dist/video-js/video.dev.js"></script>
```
Comment (dmlap) [inline test/ads.html]: Is the dev version required?

Comment (gkatsev) [inline test/ads.html]: Doesn't seem like it is here.

File: test/videojs.ads.snapshot.js [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+  };
+  player.currentTime = function(time) {
+    if (time !== undefined) {
+      currentTimeModified = true;
+    }
+  };
+
+  // with a separate video display or server-side ad insertion, ads play but
+  // the src never changes. Modifying the src or currentTime would introduce
+  // unnecessary seeking and rebuffering
```
Comment (dmlap) [inline test/videojs.ads.snapshot.js]: Will track snapshotting/restoration work correctly with stitched ads?

Comment (gkatsev) [inline test/videojs.ads.snapshot.js]: I think so. We restore tracks whether the src has changed or not.

**Pull request merged**